### PR TITLE
BC-740 - BC-854 - ansible variable generation for docker image tag

### PR DESCRIPTION
**Pull request by mamutmk5**
# Description
The ansible variable for the docker image tag will be generated by the git repository name and put to the host variables directory.

## Links to Tickets or other pull requests
https://ticketsystem.hpi-schul-cloud.org/browse/BC-854
https://ticketsystem.hpi-schul-cloud.org/browse/BC-740

## Changes
The ansible variable for the docker image tag will be generated by the git repository name and put to the host variables directory.

## Approval for review

- [x] All points were discussed with the ticket creator, support-team or product owner. The code upholds all quality guidelines from the PR-template.

> Notice: Please remove the WIP label if the PR is ready to review, otherwise nobody will review it.

### CODE REVIEW

File: .github/workflows/deploy.yml [hunk 1]
```diff
@@ -44,15 +44,20 @@ jobs:
...
           commit_id=$(git rev-parse HEAD)
           mkdir -pv ansible/group_vars/all
           for i in ./ansible/roles/*; do mkdir -pv $i/vars; done
           for i in ./ansible/roles/*; do echo "APP_IMAGE_TAG: $commit_id" >> $i/vars/main.yml; done
+          mkdir -pv ansible/host_vars/brb_host
+          mkdir -pv ansible/host_vars/default_host
+          mkdir -pv ansible/host_vars/nbc_host
+          ansible_varname=$(echo ${{ matrix.repos }} | tr [a-z] [A-Z] | tr - _ | tr \. _ | tr [:blank:] _ )
+          filename=$(echo ${ansible_varname} | tr [A-Z] [a-z] )
+          for i in ./ansible/host_vars/*; do echo "${ansible_varname}_IMAGE_TAG: $commit_id" >> $i/$filename.yml; done
```
Comment (virgilchiriac) [inline .github/workflows/deploy.yml:60]: this ticket is not about creating new files and folders, we should only do the rename

**Review by virgilchiriac — CHANGES_REQUESTED** [review]

File: .github/workflows/deploy.yml (hunk 1 above)
Comment (mamutmk5) [reply .github/workflows/deploy.yml:60]: yes and no, it's the renaming part :D
- an new variable was add, first part of renaming, for backwards compatibility the old variable and his place are in place
- the location where the variable is stored is renamed to a new location
- the way how the location where renamed is like the generation for the old location
- the new location is now the at an file at the host var folder for each host and not more in the main form the variables for the role 
- see for it https://docs.ansible.com/ansible/latest/user_guide/playbooks_variables.html#understanding-variable-precedence

Comment (alriese) [reply .github/workflows/deploy.yml:60]: Thanks, the explanation and the link was very helpful.

**Review by alriese — APPROVED** [review]

File: .github/workflows/deploy.yml (hunk 1 above)
Comment (alriese) [inline .github/workflows/deploy.yml:60]: Maybe this is easier to read:
```suggestion
          ansible_varname=$(echo ${{ matrix.repos }} | tr [a-z] [A-Z] | tr - _ | tr \. _ | tr [:blank:] _ )
          filename=$(echo ${ansible_varname} | tr [A-Z] [a-z] )
          tennens=(brb default nbc)
          for i in "${tennens[@]}"; do
                    mkdir -pv "ansible/host_vars/${i}_host"
                    echo "${ansible_varname}_IMAGE_TAG: $commit_id" >> "./ansible/host_vars/${i}_host/${filename}.yml"
          done
```

Comment (mamutmk5) [reply .github/workflows/deploy.yml:60]: no and don't make that what i whant.
Especially for what i plan later on this is not so great, for one of the next steps the folder for the host at host_vars maybe go to be dynamic.
Then the docker image tag should be go to a host explizit, that's why the host folder at the host_vars folder where create. The last thing is the merging of files who are extracted form arcives are overwriting. So it's easyser and better to put the Ansible variable with the docker image tag to seperat files.

Comment (virgilchiriac) [reply .github/workflows/deploy.yml:60]: don't understand the objection, because the suggestion does the same thing

**Review by virgilchiriac — APPROVED** [review]

**Pull request merged**